### PR TITLE
Add default value to the logprobs parameter

### LINKED ISSUE #1043: [Bug]: Missing default value to logprobs in openai.types.chat.chat_completion.Choice
### Describe the bug

https://github.com/openai/openai-python/pull/980 added token `logprobs` to chat completions of type `Optional[ChoiceLogprobs]` in [openai.types.chat.chat_completion.Choice](https://github.com/openai/openai-python/blob/3ad4e8bc9d89d7a81586bf598289ff62b0a339b9/src/openai/types/chat/chat_completion.py#L33) and [openai.types.chat.chat_completion_chunk.Choice](https://github.com/openai/openai-python/blob/3ad4e8bc9d89d7a81586bf598289ff62b0a339b9/src/openai/types/chat/chat_completion_chunk.py#L97). In the latter, the default value is set to `None`, while in the former it is not set. This causes backward compatibility problems with code written for versions before 1.5.0.

I created a https://github.com/openai/openai-python/pull/1007 in the openai repository as well, but it has to be merged here as well to ensure operability with openai versions 1.5.x and 1.6.x.

### Steps to reproduce

Tests are failing locally and in CI (e.g. https://github.com/microsoft/autogen/actions/runs/7293999612/job/19878098426)

### Expected Behavior

I made a pull request to the openai library for the parameter to have the default value set to `None`. Until then, the parameter `logprobs` should be set [here](https://github.com/microsoft/autogen/blob/a1e60e88467fef085c8417b6cef68ae172c4b520/autogen/oai/client.py#L333).

### Screenshots and logs

_No response_

### Additional Information

autogen version: 0.2.2
openai version: 1.5.0 and 1.6.0

### PR DESCRIPTION
**Pull request by davorrunje**
### Describe the bug

https://github.com/openai/openai-python/pull/980 added token `logprobs` to chat completions of type `Optional[ChoiceLogprobs]` in [openai.types.chat.chat_completion.Choice](https://github.com/openai/openai-python/blob/3ad4e8bc9d89d7a81586bf598289ff62b0a339b9/src/openai/types/chat/chat_completion.py#L33) and [openai.types.chat.chat_completion_chunk.Choice](https://github.com/openai/openai-python/blob/3ad4e8bc9d89d7a81586bf598289ff62b0a339b9/src/openai/types/chat/chat_completion_chunk.py#L97). In the latter, the default value is set to `None`, while in the former it is not set. This causes backward compatibility problems with code written for versions prior to 1.5.0.

### Steps to reproduce

Tests are failing locally and in CI (e.g. https://github.com/microsoft/autogen/actions/runs/7293999612/job/19878098426)

### Additional Information

autogen version: 0.2.2
openai version: 1.5.0 and 1.6.0

## Related issue number

Closes #1043

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by rattrayalex** [comment]
I encourage merging this; we don't anticipate adding `= None` right away.

(I'm a maintainer of the openai-python library).

**Comment by davorrunje** [comment]
> I encourage merging this; we don't anticipate adding `= None` right away.
> 
> (I'm a maintainer of the openai-python library).

Thank for the feedback :)

File: autogen/oai/client.py [hunk 1]
```diff
@@ -329,15 +330,27 @@ def _completions_create(self, client, params):
                 ),
             )
             for i in range(len(response_contents)):
-                response.choices.append(
-                    Choice(
+                try:
+                    # OpenAI versions 0.1.5 and above
```
Comment (IANTHEREAL) [inline autogen/oai/client.py:334]: 1.5.0?

Comment (davorrunje) [reply autogen/oai/client.py]: yes, fixed in the latest commit

**Review by ekzhu — APPROVED** [review]
Looks good to me

File: setup.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 __version__ = version["__version__"]
 
 install_requires = [
-    "openai>=1,<1.5",  # a temporary fix for breaking changes in 1.5
+    "openai>=1.3,<1.5",
```
Comment (sonichi) [inline setup.py:17]: shouldn't we remove "<1.5" if the PR is supposed to support openai>=1.5?

Comment (ekzhu) [reply setup.py:17]: Correct. Somehow this might have been a merge error

Comment (davorrunje) [reply setup.py:17]: Yes, that's probably due to merge. I'll run tests again against the latest 1.6.1 and create a new PR.

Comment (sonichi) [reply setup.py:17]: #1071 has been created by @ekzhu and merged.

**Pull request merged**